opendbc/safety/modes/psa.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+
+#include "opendbc/safety/safety_declarations.h"
+
+#define PSA_STEERING              757U  // RX from XXX, driver torque
+#define PSA_STEERING_ALT          773U  // RX from EPS, steering angle
+#define PSA_DYN_CMM               520U  // RX from CMM, gas pedal
+#define PSA_HS2_DYN_ABR_38D       909U  // RX from UC_FREIN, speed
+#define PSA_HS2_DAT_MDD_CMD_452   1106U // RX from BSI, cruise state
+#define PSA_DYN2_FRE              973U  // RX from ABS, brake pressure
+#define PSA_LANE_KEEP_ASSIST      1010U // TX from OP,  EPS
+
+// CAN bus
+#define PSA_CAM_BUS  0U
+#define PSA_ADAS_BUS 1U
+#define PSA_MAIN_BUS 2U
+
+static uint8_t psa_get_counter(const CANPacket_t *msg) {
+  uint8_t cnt = 0;
+  if (msg->addr == PSA_HS2_DAT_MDD_CMD_452) {
+    cnt = (msg->data[3] >> 4) & 0xFU;
+  } else if (msg->addr == PSA_HS2_DYN_ABR_38D) {
+    cnt = (msg->data[5] >> 4) & 0xFU;
+  } else {
+  }
+  return cnt;
+}
+
+static uint32_t psa_get_checksum(const CANPacket_t *msg) {
+  uint8_t chksum = 0;
+  if (msg->addr == PSA_HS2_DAT_MDD_CMD_452) {
+    chksum = msg->data[5] & 0xFU;
+  } else if (msg->addr == PSA_HS2_DYN_ABR_38D) {
+    chksum = msg->data[5] & 0xFU;
+  } else {
+  }
+  return chksum;
+}
+
+static uint8_t _psa_compute_checksum(const CANPacket_t *msg, uint8_t chk_ini, int chk_pos) {
+  int len = GET_LEN(msg);
+
+  uint8_t sum = 0;
+  for (int i = 0; i < len; i++) {
+    uint8_t b = msg->data[i];
+
+    if (i == chk_pos) {
+      // set checksum in low nibble to 0
+      b &= 0xF0U;
+    }
+    sum += (b >> 4) + (b & 0xFU);
+  }
+  return (chk_ini - sum) & 0xFU;
+}
+
+static uint32_t psa_compute_checksum(const CANPacket_t *msg) {
+  uint8_t chk = 0;
+  if (msg->addr == PSA_HS2_DAT_MDD_CMD_452) {
+    chk = _psa_compute_checksum(msg, 0x4, 5);
+  } else if (msg->addr == PSA_HS2_DYN_ABR_38D) {
+    chk = _psa_compute_checksum(msg, 0x7, 5);
+  } else {
+  }
+  return chk;
+}
+
+static void psa_rx_hook(const CANPacket_t *msg) {
+  if (msg->bus == PSA_CAM_BUS) {
+    if (msg->addr == PSA_DYN_CMM) {
+      gas_pressed = msg->data[3] > 0U; // P002_Com_rAPP
+    }
+    if (msg->addr == PSA_STEERING_ALT) {
+      int angle_meas_new = to_signed((msg->data[0] << 8) | msg->data[1], 16); // ANGLE
+      update_sample(&angle_meas, angle_meas_new);
+    }
```

Comment:
```suggestion
    if (msg->addr == PSA_DYN_CMM) {
      gas_pressed = msg->data[3] > 0U; // P002_Com_rAPP
    }

    if (msg->addr == PSA_STEERING_ALT) {
      int angle_meas_new = to_signed((msg->data[0] << 8) | msg->data[1], 16); // ANGLE
      update_sample(&angle_meas, angle_meas_new);
    }
```

I like new lines between these if statements for readability